src/bindings/python/flux/job/frobnicator/plugins/dependency.py

```diff
@@ -0,0 +1,65 @@
+##############################################################
+# Copyright 2022 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+##############################################################
+
+"""Translate dependency.name into a job id for dependency.afterok
+
+"""
+
+from flux.job.frobnicator import FrobnicatorPlugin
+
+
+class DependencyAdder:
+    """Get the system attribute for dependency.name and add a
+    task dependency. Raise an error if we can't find it.
+    """
+
+    def __init__(self, config={}):
+        # We don't need this for anything, just saving
+        # for the heck
+        self.config = config
+
+    def add_dependency(self, jobspec):
+        """We need to translate a dependency name into a job id. We will:
+        1. Get the desired name from system attribute dependency.name
+        2. list all flux jobs (I know, I know) and get the job id
+        3. Update the jobspec to have it.
+        4. Raise error if the name does not exist.
+
+        Bullet 2 is a bad design and thus this is only for experimentation.
+        """
+        dependency_name = jobspec.attributes["system"].get("dependency", {}).get("name")
+        if not dependency_name:
+            return
+
+        # We are going to add the first matched job as a dependency
+        import flux
+        import flux.job
+
+        handle = flux.Flux()
+        for job in flux.job.list.job_list(handle).get()["jobs"]:
```

Comment:
`flux.job.list.job_list()` has an optional `name` parameter you could use to get only jobs that match `dependency_name`. Also, if you just want the most recent one, I think you could also add `max_entries=1`.

Reply:
And I'm sorry, I see you put *NOT FOR REVIEW* in the PR :facepalm:

Reply:
Oh, since I already commented :slightly_smiling_face: 

You'll have to test, but `flux.job.list.job_list()` will by default only fetch jobs for the current user. In the frobnicator, this user is the owner of the instance, not necessarily the submitter of the job. You may need to set the `userid` parameter to the submission userid.

Reply:
This is super helpful! I really appreciate the feedback, regardless of my message.

Reply:
Quick question @grondo - if I added those things, would the query be efficient enough to be suitable for an actual solution (and not a hacky one I'm working on)? And more specifically, considered for adding as an optional module (not loaded by default) to the core code base?